### PR TITLE
feat(registry): contrib/registry/zookeeper add digest acl support

### PR DESCRIPTION
**Pull request by youzhixiaomutou**
#### Description (what this PR does / why we need it):

The current registrar with zookeeper doesn't have any method for auth.
I think add a simple digest acl support is necessary.

### CODE REVIEW

File: contrib/registry/zookeeper/register.go [hunk 1]
```diff
@@ -42,6 +44,20 @@ func WithTimeout(timeout time.Duration) Option {
...
+	}
+}
+
+// WithPassword with registry password.
+func WithPassword(password string) Option {
+	return func(o *options) {
+		o.password = password
+	}
+}
+
```
Comment (Cluas) [inline contrib/registry/zookeeper/register.go:60]: Do user passwords always come together, so why don't we put them together？

Comment (youzhixiaomutou) [reply contrib/registry/zookeeper/register.go:60]: like this?
```go
// WithDigestACL with registry password.
func WithDigestACL(user string, password string) Option {
	return func(o *options) {
		o.user = user
		o.password = password
	}
}
```

File: contrib/registry/zookeeper/register.go [hunk 2]
```diff
@@ -44,16 +44,10 @@ func WithTimeout(timeout time.Duration) Option {
 	return func(o *options) { o.timeout = timeout }
 }
 
-// WithUser with registry user.
-func WithUser(user string) Option {
+// WithDigestACL with registry password.
+func WithDigestACL(user string, password string) Option {
 	return func(o *options) {
 		o.user = user
```
Comment (Cluas) [inline contrib/registry/zookeeper/register.go:50]: How about merging these two directly into digestACL？

Comment (Cluas) [reply contrib/registry/zookeeper/register.go:50]: WithDigestACL build user, password to  digestACL

**Review by Cluas — APPROVED** [review]
LGTM

**Comment by Cluas** [comment]
Thanks for the contribution!

**Pull request merged**